### ISSUE TITLE
tomsguide.com

### ISSUE THREAD

**Issue by terrorist96** [uBlockOrigin/uAssets#554]
### URL(s) where the issue occurs

`https://www.tomsguide.com/us/pictures-story/919-life-improving-products.html`

### Describe the issue

Unblocked Amazon links (via `https://t.purch.com/`) and a `sols.com` ad below comments section

### Screenshot(s)

![image](https://user-images.githubusercontent.com/12802056/28392171-cc4395ce-6cae-11e7-8f1d-eea552df3ece.png)

**Comment by gorhill** [comment]
How about:

    tomsguide.com##script:inject(abort-on-property-write.js, tmnramp)

**Comment by terrorist96** [comment]
That blocks the amazon ads. However, I now see this in place of the `sols.com` ad:
![image](https://user-images.githubusercontent.com/12802056/28392494-20dc89e6-6cb0-11e7-9670-ebffe6ca44b6.png)

**Comment by gorhill** [comment]
I don't see this. Provide all the info needed for me to reproduce.

**Comment by gorhill** [comment]
How about:

    tomsguide.com##.shopping-blocks-x2

The header however is going to be a bit more work, this requires `:xpath`, not sure it's worth it.

**Comment by terrorist96** [comment]
>Never mind, I see it after refresh.

Lol now I can't see it anymore after forcing an update on my lists! Or maybe it's just cycling and it's unrelated to updating the lists. Idk haha. But this definitely works for the amazon ads:
`tomsguide.com##script:inject(abort-on-property-write.js, tmnramp)`

**Comment by terrorist96** [comment]
Actually I can now see a similar one on a different page:
`https://www.tomsguide.com/us/pictures-story/1089-best-new-iphone-games.html`
And adding `tomsguide.com##.shopping-blocks-x2` works. Thank!

**Comment by gorhill** [comment]
About `.shopping-blocks-x2`, I am not so sure these are ads. I don't that site that well, but it appears those shop boxes are all over the place on that site, including part of reviews. Example: <https://www.tomsguide.com/us/best-gaming-pc,review-2219.html>.

If the site is essentially a shopping site, I can't remove the shopping links, that is just what the site is.

**Comment by terrorist96** [comment]
Good point. But wouldn't that mean that the Amazon links should be allowed too then, since the `shopping blocks` sometimes link to Amazon as well.
Also, clicking on the `$799 Alienware` link (the first one from the link you provided), the new tab that it opens in is quickly closed by uBO. So seems like the `t.purch.com` is already somewhat blocked. Tough call.

**Comment by gorhill** [comment]
Amazon ads are clearly not part of the content, their are in a separate box with a different styles. These Amazons ads look just like the other Amazon ads on other sites.

**Comment by gorhill** [comment]
> the new tab that it opens in is quickly closed by uBO

Could be a false positive which needs to be fixed.

**Comment by terrorist96** [comment]
So then the distinguishing factor would be the presentation of the Amazon links. I'm fine with that.
